libs/x-content/src/main/java/org/elasticsearch/common/xcontent/MediaType.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.xcontent;
+
+public interface MediaType {
+    String type();
+    String subtype();
+    String format();
```

Comment:
Format, shortName, and subType seem like 3 ways to represent the same thing, and would be great if we could get to a common term (`subType`) across the board. I think we can just drop `format` in favor of `subtype` ?

Reply:
in some cases these 3 are actually different. For instance - `text/plain` has subtype=plain, but format and shortName `txt 
similarly for versioned media types I plan this to be: `application/vnd.elasticsearch+json;compatible-with=7` subtype is `vnd.elasticsearch+json`, format is `json` and shortName `json`

I think shortName is redundant and I will give it a go to remove it